monix-execution/shared/src/main/scala/monix/execution/cancelables/SingleAssignCancelable.scala

```diff
@@ -85,23 +85,26 @@ final class SingleAssignCancelable private (extra: Cancelable) extends Assignabl
     }
   }
 
-  @tailrec
   override def cancel(): Unit = {
-    state.get() match {
-      case IsCanceled | IsEmptyCanceled => ()
-      case IsActive(s) =>
-        state.set(IsCanceled)
-        if (extra != null) extra.cancel()
```

Comment:
Dotty complains about this call, which is working as intended: lampepfl/dotty#9054